rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -173,7 +173,59 @@ def _query_web3_get_logs(
     return events
 
 
-class EvmNodeInquirer(ABC):
+class UpdatableCacheDataMixin(LockableQueryMixIn):
+    def __init__(
+            self,
+            database: 'DBHandler',
+    ) -> None:
+        super().__init__()
+        self.database = database
+
+    @protect_with_lock()
+    def ensure_cache_data_is_updated(
+            self,
+            cache_type: CacheType,
+            query_method: Callable,
+            save_method: Callable,
+            force_refresh: bool = False,
+            chain_id: ChainID | None = None,
+            cache_key_parts: Sequence[str] | None = None,
+    ) -> bool:
+        """
+        It checks if the cache data is fresh enough and if not, it queries
+        the remote sources of the data and stores it to the globaldb cache tables.
+        Returns true if the cache was modified or false otherwise.
+
+        - cache type: The cache type to check for freshness
+        - query_method: The method that queries the remote source for the data
+        - save_method: The method that saves the data to the cache tables
+        - force_refresh: If True, the cache will be updated even if it is fresh
+        - cache_key_parts: The parts to be used to check cache freshness along with cache_type
+        """
+        if cache_key_parts is None:
+            cache_key_parts = []
+        if (
+            should_update_protocol_cache(cache_type, *cache_key_parts) is False and
+            force_refresh is False
+        ):
+            return False
+
+        new_data = query_method(inquirer=self, cache_type=cache_type)
+        if new_data is None:
+            return False
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            save_method_kwargs = {
+                'write_cursor': write_cursor,
+                'database': self.database,
+                'new_data': new_data,
+            }
+            if chain_id is not None:
+                save_method_kwargs['chain_id'] = chain_id
+            save_method(**save_method_kwargs)
+        return True
+
+
+class EvmNodeInquirer(ABC, UpdatableCacheDataMixin):
```

Comment:
if you do this then:

1. Is it used **anywhere** else? If not then why not just make it part of evm node inquirer itself?
2. Everywhere else that UpdatableCacheDataMixin is used should be removed. Atm every chain Inquirer inherits from it. This should be undone. Grep for uses.